src/python/pants/backend/go/goals/debug_goals.py

```diff
@@ -114,6 +121,74 @@ async def dump_go_import_paths_for_module(
     return DumpGoImportPathsForModule(exit_code=0)
 
 
+class GoExportCgoCodegenSubsystem(GoalSubsystem):
+    name = "go-export-cgo-codegen"
+    help = "Export files generated by Cgo."
+
+
+class GoExportCgoCodegen(Goal):
+    subsystem_cls = GoExportCgoCodegenSubsystem
+
+
+@goal_rule
+async def go_export_cgo_codegen(
+    targets: Targets,
+    console: Console,
+    distdir_path: DistDir,
+    workspace: Workspace,
+    golang_subsystem: GolangSubsystem,
+) -> GoExportCgoCodegen:
+    if not golang_subsystem.cgo_enabled:
+        raise ValueError("Nothing to export since cgo is disabled.")
+
+    go_package_targets = [tgt for tgt in targets if tgt.has_field(GoPackageSourcesField)]
+    for tgt in go_package_targets:
+        # Analyze the package and ensure it is actually contains cgo code.
+        analysis_wrapper = await Get(
+            FallibleFirstPartyPkgAnalysis, FirstPartyPkgAnalysisRequest(tgt.address)
+        )
```

Comment:
Ik this is a debug goal, but you can still make things much faster by using `MultiGet` outside of the for loop

Reply:
If you want to keep the code simplicity of everything happening on a single package, you can define a new rule that operates on only one package. Then, MultiGet that. So you only have one MultiGet rather than 3-4

Reply:
Done.